README


This archive contains the distribution Task-BeLike-MELO,
version 0.010:

  All the stuff I install with a fresh perl

This software is Copyright (c) 2011 by Pedro Melo.

This is free software, licensed under:

  The Artistic License 2.0 (GPL Compatible)




Changes
# Changes for Task-BeLike-MELO

{{$NEXT}}

0.011     2013-07-13 19:05:26 Europe/Lisbon
       * MooseX::AttributeHelpers is deprecated (RT86648, thx ether@cpan.org)
       * Remove some ecosystems I no longer use: Catalyst, Module::Install...
       * ... Notifo, FCGI, HTML::Template, local::lib ...
       * ... several template systems including Template Toolkit ...
       * ... XML::Compile and old ZMQ modules, IO::All insanity
       * use Method::Signatures::Simple for all my method needs
       * Settle on TOML for prefered config format, PBKDF2 for passwords
       * MSSQL is needed at work now
       * Used DBIx::Class as document store, InflateColumn::Serializer++
       * lib::tiny::findbin is gold
       * all SpreadSheet::* replaced with XLSX versions

0.010     2012-08-22 23:55:38 Europe/Lisbon
       * Added Moo, making sure we use a version that works properly
         with Moose
       * Make sure our ZMQ::LibZMQ2 is the most recent one, with the
         nasty fork() bug solved

0.009     2012-08-22 11:18:04 Europe/Lisbon
       * Code::TidyAll and friends
       * Message::Passing and friends
       * Improve Devel::Cover deps
       * Best ZeroMQ dependencies
       * other assorted tidbits

0.008     2012-06-06 11:33:48 Europe/Lisbon
       * several deps of a project I'm maintained
       * Mason, my new darling
       * Added old favorites: Try::Tiny, Sub::Name, Time::HiRes, Plack

0.007     2012-06-05 11:10:31 Europe/Lisbon
       * Moved Dist::Zilla deps to my new PluginBundle::Author::MELO
         distribution
       * Added Unicode magic from tchrist
       * Added several App:: I like a lot, including cpandoc
       * Forced to work with SOAP/WSDL, added deps for that too
       * Added Carton, to manage deps on my larger projects

0.006     2011-06-24 10:08:35 Europe/Lisbon
       * A fine CPAN module grepper command line tool
       * Make sure we have our SSL and HTTPS functions covered
       * Config::GitLike - my second choice for config files, after JSON
       * Cmd line tool uni - greps Unicode characters table
       * whiff - returns first available executable from a list
       * Remove perlbrew from our list: chicken and egg problem

0.005     2011-05-25 15:55:38 Europe/Lisbon
       * Cleanup module list: remove deprecated ones, added new ones
       * Tweak Dist::Zilla metadata

0.004 Tue Oct 19 20:15:17 UTC 2010
       * Added Throwable::X
       * Added Notifo.com modules
       * Replace plugin BumpVersionFromGit with Git::NextVersion
       * EOL tests are broken with POD sections
       * Make sure Pod::Elemental is up-to-date, required for Pod::Weaver
       * dzil plugin BumpVersionFromGit is deprecated, use Git::NextVersion now

0.003 Tue Sep 28 13:36:24 UTC 2010
      * Added deps to build the modern perl book
      * Prepare for Dist::Zilla v5

0.002 Sun Aug  1 20:42:28 UTC 2010
      * Added more modules
      * Fix .gitignore, wrong directory excluded

0.001 Sun Aug  1 20:34:15 UTC 2010
      * Initial release to CPAN


Makefile.PL

use strict;
use warnings;



use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "All the stuff I install with a fresh perl",
  "AUTHOR" => "Pedro Melo <melo\@cpan.org>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Task-BeLike-MELO",
  "EXE_FILES" => [],
  "LICENSE" => "artistic_2",
  "NAME" => "Task::BeLike::MELO",
  "PREREQ_PM" => {
    "Algorithm::C3" => "0.08",
    "AnyEvent" => "7.04",
    "AnyEvent::AIO" => "1.1",
    "AnyEvent::Gearman" => "0.09",
    "AnyEvent::HTTP" => "2.15",
    "AnyEvent::HTTPD" => "0.93",
    "AnyEvent::Redis" => "0.24",
    "AnyEvent::Twitter::Stream" => "0.22",
    "AnyEvent::WebService::Notifo" => "0.001",
    "AnyEvent::XMPP" => "0.54",
    "AnyMQ" => "0.34",
    "App::Ack" => "1.96",
    "App::Adenosine" => "2.001005",
    "App::ArchiveDevelCover" => "1.002",
    "App::BashComplete" => "0.008",
    "App::Genpass" => "2.33",
    "App::HTTPThis" => "0.002",
    "App::Nopaste" => "0.90",
    "App::Notifo" => "0.001",
    "App::Uni" => 8,
    "App::Whiff" => "0.002",
    "App::autotest" => "0.004",
    "App::cpanminus" => "1.5019",
    "App::cpanoutdated" => "0.24",
    "App::p" => "0.0303",
    "App::pod2pdf" => "0.42",
    "Archive::Tar" => "1.90",
    "Archive::Tar::Wrapper" => "0.16",
    "Archive::Zip" => "1.30",
    "Async::Hooks" => "0.16",
    "Authen::Simple::Passwd" => "0.6",
    "CGI::Emulate::PSGI" => "0.15",
    "CGI::Fast" => "1.09",
    "CHI" => "0.56",
    "CPAN" => "1.9800",
    "CPAN::Mini" => "1.111010",
    "CPAN::Mini::Webserver" => "0.58",
    "CPAN::Reporter" => "1.2009",
    "CPAN::SQLite" => "0.202",
    "CSS::Minifier" => "0.01",
    "Cache::CacheFactory" => "1.10",
    "Carp::Always" => "0.12",
    "Carp::Clan" => "6.04",
    "Carton" => "v0.9.4",
    "Catalyst::Action::RenderView" => "0.16",
    "Catalyst::Controller::REST" => "1.06",
    "Catalyst::Devel" => "1.37",
    "Catalyst::Engine::PSGI" => "0.13",
    "Catalyst::Log::Log4perl" => "1.06",
    "Catalyst::Plugin::Authentication" => "0.10021",
    "Catalyst::Plugin::Session" => "0.36",
    "Catalyst::Plugin::Session::State::Cookie" => "0.17",
    "Catalyst::Plugin::Session::Store::FastMmap" => "0.16",
    "Catalyst::Plugin::StackTrace" => "0.11",
    "Catalyst::Plugin::Unicode" => "0.93",
    "Catalyst::Runtime" => "5.90019",
    "Catalyst::View::TT" => "0.39",
    "Catalyst::View::Tenjin" => "0.050001",
    "Check::ISA" => "0.04",
    "Class::C3" => "0.24",
    "Class::C3::XS" => "0.13",
    "Class::Load" => "0.20",
    "Class::MOP" => "2.0604",
    "Class::XSAccessor" => "1.16",
    "Clipboard" => "0.13",
    "Code::TidyAll" => "0.16",
    "Code::TidyAll::Plugin::Perl::AlignMooseAttributes" => "0.01",
    "Code::TidyAll::Plugin::Perl::IgnoreMethodSignaturesSimple" => "0.02",
    "Code::TidyAll::Plugin::PerlTidy" => "0.16",
    "Code::TidyAll::Plugin::PodTidy" => "0.16",
    "Compress::Raw::Bzip2" => "2.059",
    "Compress::Raw::Zlib" => "2.059",
    "Compress::Zlib" => "2.059",
    "Config::Any" => "0.23",
    "Config::GitLike" => "1.10",
    "Coro" => "6.23",
    "Crypt::Eksblowfish::Bcrypt" => "0.009",
    "Crypt::SSLeay" => "0.64",
    "DBD::SQLite" => "1.37",
    "DBD::mysql" => "4.022",
    "DBI" => "1.623",
    "DBICx::DataDictionary" => "0.002",
    "DBICx::Indexing" => "0.002",
    "DBICx::Shortcuts" => "0.007",
    "DBIx::Class" => "0.08204",
    "DBIx::Class::Candy" => "0.002103",
    "DBIx::Class::DeploymentHandler" => "0.002203",
    "DBIx::Class::EncodedColumn" => "0.00011",
    "DBIx::Class::Helpers" => "2.016003",
    "DBIx::Class::InflateColumn::Currency" => "0.02005",
    "DBIx::Class::InflateColumn::URI" => "0.01002",
    "DBIx::Class::Schema::Loader" => "0.07033",
    "DBIx::Class::TimeStamp" => "0.14",
    "DBIx::Connector" => "0.52",
    "DBIx::Tracer" => "0.02",
    "Data::Dump" => "1.21",
    "Data::Dumper" => "2.139",
    "Data::ObjectDriver" => "0.09",
    "Data::UUID::MT" => "1.000",
    "Date::Manip" => "6.37",
    "Date::Tiny" => "1.04",
    "DateTime" => "0.78",
    "DateTime::Format::DateParse" => "0.05",
    "DateTime::Format::Excel" => "0.31",
    "DateTime::Format::MySQL" => "0.04",
    "DateTime::Format::SQLite" => "0.11",
    "Devel::BeginLift" => "0.001003",
    "Devel::Cycle" => "1.11",
    "Devel::LeakGuard::Object" => "0.06",
    "Devel::NYTProf" => "4.23",
    "Devel::REPL" => "1.003014",
    "Devel::StackTrace" => "1.30",
    "Devel::StackTrace::WithLexicals" => "0.10",
    "Devel::TraceUse" => "2.08",
    "Digest::HMAC" => "1.03",
    "Digest::MD5" => "2.52",
    "Digest::SHA" => "5.80",
    "Digest::SHA1" => "2.13",
    "Dist::Zilla" => "4.300028",
    "Dist::Zilla::App::Command::cover" => "1.101001",
    "Dist::Zilla::Plugin::InstallGuide" => "1.200000",
    "Dist::Zilla::Plugin::LatestPrereqs" => "0.4",
    "Dist::Zilla::Plugin::PodPurler" => "0.093400",
    "Dist::Zilla::PluginBundle::Author::MELO" => "0.011",
    "EV" => "4.11",
    "Email::Abstract" => "3.004",
    "Email::Simple" => "2.102",
    "Eval::Closure" => "0.08",
    "ExtUtils::CBuilder" => "0.280205",
    "ExtUtils::MakeMaker" => "6.64",
    "FCGI" => "0.74",
    "FCGI::Client" => "0.08",
    "FCGI::ProcManager" => "0.24",
    "File::ChangeNotify" => "0.22",
    "File::HomeDir" => "1.00",
    "File::HomeDir::PathClass" => "1.112060",
    "File::MMagic" => "1.29",
    "File::Spec" => "3.33",
    "File::Temp" => "0.22",
    "File::Which" => "1.09",
    "Format::Human::Bytes" => "0.06",
    "GD" => "2.46",
    "Gearman::Client" => "1.11",
    "Gearman::Worker" => 0,
    "Getopt::Long" => "2.38",
    "Github::Import" => "0.07",
    "Graph::Easy" => "0.72",
    "GraphViz" => "2.14",
    "Guard" => "1.022",
    "HTML::Mason" => "1.50",
    "HTML::Template" => "2.91",
    "HTML::Template::Expr" => "0.07",
    "HTML::Template::JIT" => "0.05",
    "HTML::Template::Pro" => "0.9509",
    "HTTP::Headers" => "6.05",
    "HTTP::Parser" => "0.06",
    "HTTP::Parser::XS" => "0.15",
    "HTTP::Tiny" => "0.025",
    "IO::AIO" => "4.18",
    "IO::All" => "0.46",
    "IO::Compress::Base" => "2.059",
    "IO::Handle::Util" => "0.01",
    "IO::Socket::SSL" => "1.81",
    "IO::Uncompress::Gunzip" => "2.059",
    "IO::Zlib" => "1.10",
    "Image::Size" => "3.232",
    "JSON::XS" => "2.33",
    "KiokuDB" => "0.52",
    "KiokuDB::Backend::DBI" => "1.22",
    "KiokuDB::Backend::Redis" => "0.02",
    "LWP::Protocol::https" => "6.03",
    "LWP::Simple" => "6.00",
    "List::Util" => "1.27",
    "Log::Any" => "0.14",
    "Log::Any::Adapter" => "0.11",
    "Log::Any::Adapter::Log4perl" => "0.06",
    "Log::Log4perl" => "1.40",
    "MIME::Base64" => "3.09",
    "MIME::Entity" => "5.503",
    "MIME::Lite" => "3.029",
    "MIME::Parser" => "5.503",
    "MIME::WordDecoder" => 0,
    "MLDBM" => "2.04",
    "Mail::Address" => "2.12",
    "Mail::IMAPClient" => "3.32",
    "Markapl" => "0.14",
    "Mason" => "2.20",
    "Mason::Tidy" => "2.57",
    "Math::Round" => "0.06",
    "Message::Passing" => "0.110",
    "Message::Passing::Redis" => "0.006",
    "Message::Passing::ZeroMQ" => "0.005",
    "Method::Signatures" => 20121201,
    "Modern::Perl" => "1.20121103",
    "Module::Build" => "0.4003",
    "Module::CoreList" => "2.79",
    "Module::Info" => "0.32",
    "Module::Install" => "1.06",
    "Module::Install::AuthorTests" => "0.002",
    "Module::Install::AutoManifest" => "0.003",
    "Module::Install::ProvidesClass" => "1.000000",
    "Module::Install::ReadmeFromPod" => "0.20",
    "Module::Install::Repository" => "0.06",
    "Module::Install::XSUtil" => "0.43",
    "Module::Which" => "0.0207",
    "Mojo" => 0,
    "Moo" => "1.000003",
    "Moose" => "2.0604",
    "Moose::Exporter" => "2.0604",
    "MooseX::Aliases" => "0.10",
    "MooseX::AttributeHelpers" => "0.23",
    "MooseX::AttributeShortcuts" => "0.017",
    "MooseX::ConfigFromFile" => "0.04",
    "MooseX::Declare" => "0.35",
    "MooseX::Getopt" => "0.50",
    "MooseX::Has::Sugar" => "0.05070421",
    "MooseX::HasDefaults::RO" => 0,
    "MooseX::LazyRequire" => "0.09",
    "MooseX::Method::Signatures" => "0.44",
    "MooseX::MethodAttributes" => "0.28",
    "MooseX::MultiMethods" => "0.10",
    "MooseX::Role::WithOverloading" => "0.12",
    "MooseX::StrictConstructor" => "0.19",
    "MooseX::Traits" => "0.11",
    "MooseX::Types" => "0.35",
    "MooseX::Types::DateTime" => "0.08",
    "MooseX::Types::Moose" => "0.35",
    "Mozilla::CA" => 20120823,
    "NTS::Template" => "2.1",
    "Net::Bonjour" => "0.96",
    "Net::Cmd" => "2.29",
    "Net::DNS" => "0.72",
    "Net::FTP" => "2.77",
    "Net::Rendezvous" => "0.92",
    "Net::Rendezvous::Publish::Backend::Apple" => "0.02",
    "Net::SMTP" => "2.31",
    "Net::SMTP::TLS::ButMaintained" => "0.21",
    "Net::SSLeay" => "1.51",
    "Object::Event" => "1.23",
    "POE" => "1.354",
    "POE::XS::Queue::Array" => "0.006",
    "PPI" => "1.215",
    "PPI::HTML" => "1.08",
    "Package::Stash" => "0.34",
    "Package::Stash::XS" => "0.26",
    "Parallel::Prefork" => "0.13",
    "Params::Coerce" => "0.14",
    "Parse::CPAN::Meta" => "1.4404",
    "Parse::RecDescent" => "1.967009",
    "Path::Class" => "0.29",
    "Perl::Critic" => "1.118",
    "Perl::PrereqScanner" => "1.015",
    "Perl::Tidy" => 20121207,
    "Plack" => "1.0014",
    "Plack::Middleware::Debug" => "0.14",
    "Plack::Middleware::Debug::DBIC::QueryLog" => "0.09",
    "Plack::Middleware::Debug::DBIProfile" => "0.102",
    "Plack::Server::ReverseHTTP" => "0.03",
    "Pod::Coverage::TrustPod" => "0.100002",
    "Pod::Cpandoc" => "0.13",
    "Pod::Elemental" => "0.102362",
    "Pod::POM::Web" => "1.18",
    "Pod::PseudoPod" => "0.18",
    "Pod::Tidy" => "0.10",
    "Poet" => "0.13",
    "Redis" => "1.955",
    "Scalar::Util" => "1.27",
    "Server::Starter" => "0.12",
    "Spreadsheet::ParseExcel" => "0.59",
    "Spreadsheet::WriteExcel" => "2.38",
    "String::CRC32" => "1.4",
    "String::FriendlyID" => "0.019",
    "Sub::Name" => "0.05",
    "Sys::UniqueID" => "1.0",
    "TAP::Formatter::HTML" => "0.09",
    "TAP::Formatter::TextMate" => "0.1",
    "Task::Devel::Cover::Recommended" => "0.940000",
    "Task::Kensho" => "0.33",
    "Task::Moose" => "0.03",
    "Task::Plack" => "0.26",
    "Template" => "2.24",
    "Template::Alloy" => "1.016",
    "Template::Declare" => "0.46",
    "Template::Parser::CET" => "0.05",
    "Template::Sandbox" => "1.04",
    "Template::Tiny" => "1.12",
    "Tenjin" => "0.070001",
    "Term::ProgressBar" => "2.13",
    "Term::ReadKey" => "2.30",
    "Term::ReadLine::Gnu" => "1.20",
    "Term::ReadLine::Perl" => "1.0303",
    "Test::Base" => "0.60",
    "Test::CPAN::Meta" => "0.22",
    "Test::Compile" => "0.22",
    "Test::Continuous" => "0.73",
    "Test::Deep" => "0.110",
    "Test::Exception" => "0.31",
    "Test::Fatal" => "0.010",
    "Test::Harness" => "3.25",
    "Test::Inline::Extract" => "2.212",
    "Test::Kit" => "0.101",
    "Test::More" => "0.98",
    "Test::Output" => "1.01",
    "Test::Reporter" => "1.58",
    "Test::Reporter::Transport::Metabase" => "1.999008",
    "Test::TCP" => "1.18",
    "Test::WWW::Mechanize::Catalyst" => "0.58",
    "Text::Autoformat" => "1.669002",
    "Text::ClearSilver" => "0.010005004",
    "Text::Glob" => "0.09",
    "Text::Markdown" => "1.000031",
    "Text::MicroMason" => "2.13",
    "Text::MultiMarkdown" => "1.000034",
    "Text::ParseWords" => "3.27",
    "Text::Template::Simple" => "0.86",
    "Text::Textile" => "2.12",
    "Text::Tmpl" => "0.33",
    "Text::Unaccent" => "1.08",
    "Text::Wrap" => "2012.0818",
    "Text::Xslate" => "1.6002",
    "TextMate::JumpTo" => "0.07",
    "Throwable::X" => "0.006",
    "Tie::IxHash" => "1.22",
    "Time::HiRes" => "1.9725",
    "Time::ParseDate" => "2011.0517",
    "Time::y2038" => 20100403,
    "Try::Tiny" => "0.12",
    "TryCatch" => "1.003000",
    "Twiggy" => "0.1021",
    "URI" => "1.60",
    "URI::Escape" => "3.31",
    "URI::Find" => 20111103,
    "Unicode::String" => "2.09",
    "Unicode::Stringprep" => "1.104",
    "Unicode::Tussle" => "1.03",
    "Web::Scraper" => "0.37",
    "XML::Compile" => "1.30",
    "XML::Compile::WSDL11" => "2.34",
    "XML::Feed" => "0.51",
    "XML::LibXML" => "2.0014",
    "XML::Parser" => "2.41",
    "XML::Parser::Expat" => "2.41",
    "XML::Simple" => "2.20",
    "XML::XPath" => "1.13",
    "XS::Assert" => "0.002",
    "YAML" => "0.84",
    "YAML::XS" => "0.38",
    "ZMQ" => "1.05",
    "ZMQ::LibZMQ3" => "1.03",
    "aliased" => "0.30",
    "autodie" => "2.13",
    "lib::tiny" => "0.7",
    "lib::xi" => "0.13",
    "local::lib" => "1.008004",
    "namespace::autoclean" => "0.13",
    "namespace::clean" => "0.24"
  },
  "VERSION" => "0.011",
  "test" => {
    "TESTS" => ""
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



